pkg/pipelines/tekton_checks.go

```diff
@@ -0,0 +1,91 @@
+package pipelines
+
+import (
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// getCrdInterface returns CustomResourceDefinitionInterface
+func getCrdInterface() (v1beta1.CustomResourceDefinitionInterface, error) {
+	// obtain kubeconfig
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	clientConfig, err := kubeconfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	// get client set from client config
+	cs, err := clientset.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return cs.ApiextensionsV1beta1().CustomResourceDefinitions(), nil
+}
+
+// isCrdFound retuns true if crdName is found.  Otherwise, it returns false.
+func isCrdFound(crdInterface v1beta1.CustomResourceDefinitionInterface, crdName string) (bool, error) {
+	crd, err := crdInterface.Get(crdName, metav1.GetOptions{})
+	if err != nil {
+		// return false if crd is not found
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	if crd == nil {
+		return false, nil
```

Comment:
It's pretty rare to use `else` in Go.

This could be shortened to `return crd != nil, nil`